chain-impl-mockchain/src/update.rs

```diff
@@ -101,7 +103,9 @@ impl UpdateState {
                 // If a majority of BFT leaders voted for the
                 // proposal, then apply it.
                 if proposal_state.votes.size() > settings.bft_leaders.len() / 2 {
-                    settings = settings.apply(proposal_state.proposal.changes())?;
+                    settings = settings
+                        .try_apply(proposal_state.proposal.changes())
+                        .unwrap();
```

Comment:
I would add a panic warning in the function documentation

Reply:
updated